salt/modules/zypper.py

```diff
@@ -357,10 +357,12 @@ def info_installed(*names, **kwargs):
         t_nfo = dict()
         # Translate dpkg-specific keys to a common structure
         for key, value in pkg_nfo.items():
-            if six.PY2 and type(value) == str:
+            if type(value) == str:
```

Comment:
Oh, also note that when using `six.string_types`, python 2 uses `basestring` like you mentioned:
https://github.com/saltstack/salt/blob/1f989e720a65657b34a5a4aa3c1bae4fadd57ef8/salt/ext/six.py#L55

I've updated the first `if` statement with the `six.string_types` check in #35181.